src/io/register_buffers.rs

```diff
@@ -5,24 +5,24 @@ use std::cell::RefCell;
 use std::io;
 use std::rc::Rc;
 
-pub(crate) fn register_buffers(buffers: &Rc<RefCell<FixedBuffers>>) -> io::Result<()> {
+pub(crate) fn register_buffers(buffers: Rc<RefCell<dyn FixedBuffers>>) -> io::Result<()> {
```

Comment:
As explained in the previous comment, we cannot borrow a reference to a fat `Rc<RefCell<dyn FixedBuffers>>` pointer, without that fat pointer being instantiated somewhere, and the `Rc<RefCell<Inner>>` member of the collection struct is not it. We have to first clone and coerce it to a temporary fat pointer, and since we have done that, why not spare an extra indirection and pass the temporary `Rc` in by value?